lib/spack/spack/test/cmd/env.py

```diff
@@ -2002,3 +2003,36 @@ def test_env_write_only_non_default():
         yaml = f.read()
 
     assert yaml == ev.default_manifest_yaml
+
+
+def test_update_and_revert_anonymous_env(tmpdir):
+    raw_yaml = """
+spack:
+  specs:
+  - mpich
+  packages:
+    cmake:
+      paths:
+        cmake@3.17.3: /usr
+"""
+    manifest = tmpdir.ensure('spack.yaml')
+    backup_file = tmpdir.join('spack.yaml.bkp')
+    manifest.write(raw_yaml)
+    env('update', '-y', str(manifest.dirname))
+
+    # Check if the backup file has been created
+    assert os.path.exists(str(backup_file))
+    # Try to read the environment (it should not error)
+    ev.create('test', str(manifest))
+
+    # Retrying another update does nothing since the
+    # manifest is up-to-date
+    env('update', '-y', str(manifest.dirname))
```

Comment:
I think what you are testing here is whether you can update something twice. I think that:

* you should state that explicitly
* it might make sense to split off into a separate unit test (i.e. have a dedicated unit test to make sure that updating twice works)

Reply:
Done in 05c99a0